### PR TITLE
[ISSUE-307] Prepare CRD for CSI deployment

### PR DESCRIPTION
**Pull request by AnastasiaShemyakinskaya**
### Issue #307 

* Add proposal with CRD structure for CSI

## PR checklist
- [x] Add link to the issue
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
_Link to custom CI build_

### CODE REVIEW

File: docs/proposals/csi-deployment-crd.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+metadata:
+  name: csi.dell.com
+  labels:
+    app.kubernetes.io/name: csi-operator
+spec:
+  group: csi.dell.com
+  names:
+    kind: CSI
+    listKind: CSIList
+    plural: csi
```
Comment (mishoyama) [inline docs/proposals/csi-deployment-crd.md:28]: Is it possible to limit number of CR in CSI CRD?

File: docs/proposals/csi-deployment-crd.md [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+Currently CSI is deployed using helm charts and value
+
+## Proposal
+
+Example of CRD:
+```
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: csi.dell.com
```
Comment (mishoyama) [inline docs/proposals/csi-deployment-crd.md:20]: `csi-baremetal.dell.com` I guess

File: docs/proposals/csi-deployment-crd.md [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+Example of CR:
+```
+apiVersion: csi.dell.com/v1beta1
+kind: CSI
+metadata:
+  name: csi
+  namespace: default
+  labels:
+    app.kubernetes.io/name: csi-baremetal
+    app.kubernetes.io/managed-by: csi-operator
```
Comment (mishoyama) [inline docs/proposals/csi-deployment-crd.md:46]: `csi-baremetal-operator`

File: docs/proposals/csi-deployment-crd.md [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+    app.kubernetes.io/version: 1.0.0
+spec:
+  driver:
+    controller:
+      image:
+        registry: asdrepo.isus.emc.com:9042
```
Comment (mishoyama) [inline docs/proposals/csi-deployment-crd.md:52]: Can we have global setting for registry?

File: docs/proposals/csi-deployment-crd.md [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+    - name: v1beta1
+      served: true
+      storage: true
+```
+
+Example of CR:
+```
+apiVersion: csi.dell.com/v1beta1
+kind: CSI
+metadata:
```
Comment (mishoyama) [inline docs/proposals/csi-deployment-crd.md:41]: We also need to specify node selectors and ability to not deploy scheduler extender, don't manage nodes CRD, etc.

**Review by mishoyama — CHANGES_REQUESTED** [review]

File: docs/proposals/csi-deployment-crd.md [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+      port: 8787
+    logReceiver:
+      name: fluent-bit
+      image:
+        name: fluent-bit
+        registry: asdrepo.isus.emc.com:9042
+        pullPolicy: Always
+        tag: shippable
+    deployAlertsConfig: false
+    nodeSelector:
```
Comment (mishoyama) [inline docs/proposals/csi-deployment-crd.md:116]: `nodeSelector` should be a common setting. Is `os=linux` just an example? We can't have default value here...

Comment (AnastasiaShemyakinskaya) [reply docs/proposals/csi-deployment-crd.md:116]: It's just an example

File: docs/proposals/csi-deployment-crd.md (hunk 1 above)
Comment (AnastasiaShemyakinskaya) [reply docs/proposals/csi-deployment-crd.md:28]: I didn't find such information. It's seems like CRD specification doesn't contain special field for this case. I suppose we can only handle it in operator code

**Review by mishoyama — APPROVED** [review]

File: docs/proposals/csi-deployment-crd.md [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+  namespace: default
+  labels:
+    app.kubernetes.io/name: csi-baremetal
+    app.kubernetes.io/managed-by: csi-baremetal-operator
+    app.kubernetes.io/version: 1.0.0
+spec:
+  driver:
+    controller:
+      image:
+        registry: asdrepo.isus.emc.com:9042
```
Comment (mishoyama) [inline docs/proposals/csi-deployment-crd.md:52]: Please remove reference to proprietary registry

**Review by safronovD — APPROVED** [review]

**Pull request merged**